verification/certifyKey.go

```diff
@@ -433,9 +433,11 @@ func checkCertificateStructure(t *testing.T, certBytes []byte) *x509.Certificate
 
 func testCertifyKey(d TestDPEInstance, client DPEClient, t *testing.T, simulation bool) {
 	ctx := getInitialContextHandle(d, client, t, simulation)
-	if simulation {
-		defer client.DestroyContext(ctx, 0)
-	}
+	defer func() {
+		if simulation {
+			client.DestroyContext(ctx, 0)
+		}
+	}()
```

Comment:
- Yes, they are different especially in Simulation mode. 
- If defer "function call" is used instead of defer "function block", it causes error and subsequent Go tests break with "Invalid handle" error.
- This is because some "older" context handle is passed to DestroyContext() instead of "newest" context handle.  

#### **Here is why:**
    - In simulation mode, the "ctx" value keeps changing unlike default context (because CertifyKey in default context will return default handle only.)
    - The problem arises due to incorrect evaluation & assignment to "ctx" variable.
    - When we use defer, we expect that "ctx" must hold the new context handle.
    - But, it behaves differently for "function call" and "anonymous function call" as below.
    - **In case of deferred function call,** even though the DestroyContext() is call is deferred, the value of "ctx" is assigned beforehand. So, DestroyContext() tries to delete an older context (that doesn't even exist)  - This breaks subsequent tests with "Invalid handle error"
    - But **in case of deferred anonymous function call,**   the anonymous function block is evaluated at the end only. "ctx" variable has the "newest" context handle. DestroyContext() works as expected.
    https://stackoverflow.com/a/47321051